[ci-visibility] Add Javascript Agentless Instructions (#13771)
DataDog/documentation: content/en/continuous_integration/setup_tests/javascript.md

[Install the Datadog Agent to collect tests data][4].

<div class="alert alert-warning">

> Agentless mode is in beta. To test this feature, follow the <a href="/continuous_integration/setup_tests/javascript#agentless-beta">instructions</a> on this page.

Reviewer: It looks like you can add relative links into the hyperlink of the banner, which I'm surprised to learn about today!

```suggestion
Agentless mode is in beta. To test this feature, follow the <a href="/continuous_integration/setup_tests/javascript#agentless-beta">Agentless instructions</a> on this page.
```

: Commit committer date in ISO 8601 format.<br/>
**Example**: `2021-03-12T16:00:28Z`

## Agentless (Beta)

To instrument your test suite without requiring an Agent, configure the following environment variables:

`DD_CIVISIBILITY_AGENTLESS_ENABLED` (Required)
: Enables or disables Agentless mode.<br/>
**Default**: `false`

`DD_API_KEY` (Required)
: The [Datadog API key][8] used to upload the test results.<br/>
**Default**: `(empty)`

> Additionally, configure which [Datadog site][7] to which you want to send data. Your Datadog site is: {{< region-param key="dd_site" >}}.

Reviewer: ```suggestion
Additionally, configure which [Datadog site][7] you want to send data to. Your Datadog site is: {{< region-param key="dd_site" >}}.
```

`DD_SITE` (Required)
: The [Datadog site][9] to upload results to.<br/>
**Default**: `datadoghq.com`<br/>
**Selected site**: {{< region-param key="dd_site" code="true" >}}

## Known limitations

### ES modules
[Mocha >=9.0.0][10] uses an ESM-first approach to load test files. That means that if ES modules are used (for example, by defining test files with the `.mjs` extension), _the instrumentation is limited_. Tests are detected, but there isn't visibility into your test. For more information about ES modules, see the [NodeJS documentation][11].

### Browser tests
Browser tests executed with `mocha`, `jest`, `cucumber` and `cypress` are instrumented by `dd-trace-js`, but visibility into the browser session itself is not provided by default (for example, network calls, user actions, page loads, and so on).

> If you want visibility into the browser process, consider using [Real User Monitoring][12]. When using Cypress, test results and their generated RUM browser sessions and session replays are automatically linked. Learn more in the [RUM integration][13] guide.

Reviewer: ```suggestion
If you want visibility into the browser process, consider using [RUM & Session Replay][12]. When using Cypress, test results and their generated RUM browser sessions and session replays are automatically linked. Learn more in the [RUM integration][13] guide.
```
